datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -390,6 +390,22 @@ impl<'a, S: SimplifyInfo> ExprRewriter for Simplifier<'a, S> {
                 lit(negated)
             }
 
+            // expr IN ((subquery)) -> expr IN (subquery), see ##5529
+            Expr::InList {
+                expr,
+                mut list,
+                negated,
+            } if list.len() == 1
+                && matches!(list.first(), Some(Expr::ScalarSubquery { .. })) =>
+            {
+                let Expr::ScalarSubquery(subquery) = list.remove(0) else { unreachable!() };
```

Comment:
Struggling to get why we unreachable here

Reply:
The condition in L399 ensures that only `ScalarSubquery` is allowed, making other branches unreachable. Although there is an unstable [rust feature](https://github.com/rust-lang/rust/issues/51114) available to simplify this, it cannot be used at the moment.